volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -2620,14 +2631,30 @@ def get_content(self) -> Union[str, None]:
         """
         vmlinux = linux.LinuxUtilities.get_module_from_volobj_type(self._context, self)
         vmlinux_layer = vmlinux.context.layers[vmlinux.layer_name]
-        physical_layer = vmlinux.context.layers["memory_layer"]
+        physical_layer_name = self._context.layers[self.vol.layer_name].config.get(
+            "memory_layer", self.vol.layer_name
```

Comment:
Need to be really careful about this.  I feel at some point soon we're going to have to go back and find all these instances that get the physical layer that way (this ignores swap space and hardwires the expectation of a `memory_layer` into the code, which is only on intel and could theoretically change in the future).  Technically it should go through the `.dependencies` of the layer and choose one of them (somehow).  I won't push that at the moment, and I think we could find all instances of this by looking for the string `memory_layer`, just putting in a heads-up here...